packages/flutter_tools/lib/src/test/flutter_tester_device.dart

```diff
@@ -176,25 +174,23 @@ class FlutterTesterTestDevice extends TestDevice {
         }
 
         logger.printTrace('Connecting to service protocol: $forwardingUri');
-        final Future<FlutterVmService> localVmService = connectToVmService(
+        final FlutterVmService vmService = await connectToVmServiceImpl(
```

Comment:
Can you write a regression test? Maybe by making the anonymous closure starting on line 158 a `@visibleForTesting` method?

Reply:
[This test](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/integration.shard/test_test.dart#L216) should already cover this (it was the test that was failing when [this CL disabling Observatory by default](https://github.com/dart-lang/sdk/commit/fb1516c4ea52111f05a96d9eaedffc1e6feed000A) landed in the SDK).

Reply:
LGTM